### PR TITLE
Map update-state flags

### PR DESCRIPTION
**Pull request by hjanetzek**
Instead of returning kind of arbitrary bool from `Map::update()` and `Map::render()` this returns a bitset with all the state flags from `Map::update()`

### CODE REVIEW

**Comment by tallytalwar** [comment]
Nice.. code looks clean. Will give it a try on all platforms and merge.

File: core/include/tangram/map.h [hunk 1]
```diff
@@ -185,13 +202,14 @@ class Map {
     // Resize the map view to a new width and height (in pixels)
     void resize(int _newWidth, int _newHeight);
 
-    // Update the map state with the time interval since the last update, returns
+    // Update the map state with the time interval since the last update.
+    //returns
     // true when the current view is completely loaded (all tiles are available and
     // no animation in progress)
-    bool update(float _dt);
```
Comment (hjanetzek) [inline core/include/tangram/map.h]: Need to update docs

**Comment by matteblair** [comment]
This change makes the macOS app completely unresponsive - not sure why yet.

**Comment by hjanetzek** [comment]
Seeing this on linux as well. gdb attached shows `visibleTiles = std::set with 31416 elements` Checking further tomorrow

**Comment by matteblair** [comment]
Yep, I'm seeing a flood of tile load tasks on macOS - seems to only happen when the scene tries to prefetch tiles.

**Comment by hjanetzek** [comment]
Oh yes.. with the `view->update()` in `Map::update()` while scene is loading (and prefetching) `view` could have been modified concurrently.

**Comment by matteblair** [comment]
Looks like it's fixed now - thanks!

File: platforms/android/tangram/src/main/java/com/mapzen/tangram/MapRenderer.java [hunk 2]
```diff
@@ -34,29 +41,41 @@ public void onDrawFrame(final GL10 gl) {
...
+            if (!isPrevCameraEasing) {
+                uiThreadHandler.post(setMapRegionAnimatingRunnable);
+            }
         } else if (isPrevCameraEasing) {
             uiThreadHandler.post(setMapRegionIdleRunnable);
         }
 
-        boolean viewComplete = mapViewComplete && !isPrevMapViewComplete;
+        if (isAnimating) {
+            map.requestRender();
```
Comment (matteblair) [inline platforms/android/tangram/src/main/java/com/mapzen/tangram/MapRenderer.java]: It seems simpler to apply the re-rendering logic in the core, rather than duplicating it for each of the platforms. Why this change?

Comment (hjanetzek) [reply platforms/android/tangram/src/main/java/com/mapzen/tangram/MapRenderer.java]: It's mostly beneficial on Android to safe one native-to-java call. Though to me this seems also is more clean - or otherwise name it `updateAndRequestRerenderAsNeeded()`

Comment (matteblair) [reply platforms/android/tangram/src/main/java/com/mapzen/tangram/MapRenderer.java]: Fair enough!
